currency/pair_test.go

```diff
@@ -397,7 +397,7 @@ func TestNewPairDelimiter(t *testing.T) {
 		t.Fatal(err)
 	}
 	actual = pair.String()
-	expected = "fBTC-USDT"
+	expected = "fbtc-USDT"
```

Comment:
Well how the currency code matches it could happen via GCTCLI. If we go BTC-usdt it will have that case, as long as we format it correctly it's good to go. Internally no, but with external funniness possibly.

Reply:
But you do uppercase matching
```
	// Force upper string storage and matching
	c = strings.ToUpper(c)
```
Its more that there is a conscious choice to return a `Code` as upper or lower, but then the currency.Pair doesn't mind that it has two formats

Reply:
So upping all strings consolidates currency Item struct generation in memory, so btc == bTc == BTC instead of appending more of the same and having a hard time matching with a hash lookup which utilises the currency items pointer just so we can ensure commonality across the instance. The code struct will have the format determined by the initial first case char on currency code. If you think there is a better way design-wise, happy to sit down discuss a change.